src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -274,11 +276,26 @@ public CompiledScript compileInternal(Script script) {
         CompiledScript compiled = cache.getIfPresent(cacheKey);
         if (compiled == null) {
             //Either an un-cached inline script or an indexed script
-            compiled = new CompiledScript(lang, scriptEngineService.compile(code));
+            try {
+                compiled = new CompiledScript(script.getType(), name, lang, scriptEngineService.compile(code));
+            } catch (RuntimeException re) {
+                throw new RuntimeException("Compilation error with " + script.getType().name()
```

Comment:
Oops.  I missed that.  This is a better way to do this, and I will make the change.